src/audio/kpb.c

```diff
@@ -2060,6 +2068,50 @@ static inline void kpb_change_state(struct comp_data *kpb,
 	kpb->state_log = (kpb->state_log << 4) | state;
 }
 
+static int kpb_set_micselect(struct comp_dev *dev, const void *data,
+			     int max_data_size)
+{
+	comp_info(dev, "kpb_set_micselect");
+	const struct kpb_micselector_config *mic_sel = data;
+	struct comp_data *kpb = comp_get_drvdata(dev);
+	const size_t mic_cnt = kpb->config.channels - KPB_REFERENCE_SUPPORT_CHANNELS;
```

Comment:
Is this guaranteed config.channels is at least KBP_REFERENCE_SUPPORT_CHANNELS so that this doesn't become negative?

Reply:
Yes. KPB gets a minimum of 3 channels.